packages/go_router/lib/src/route.dart

```diff
@@ -5,32 +5,22 @@
 import 'package:collection/collection.dart';
 import 'package:flutter/widgets.dart';
 
+import 'configuration.dart';
+import 'pages/custom_transition_page.dart';
 import 'path_utils.dart';
-import 'state.dart';
 import 'typedefs.dart';
 
-/// A declarative mapping between a route path and a page builder.
-class GoRoute {
-  /// Default constructor used to create mapping between a route path and a page
-  /// builder.
-  GoRoute({
+/// The base class for all route configuration objects.
```

Comment:
Maybe add some breadcumb to different subclasses and some more description on when would user want to use each of the subclass